### PR TITLE
aws - kinesis - cross account filter

### PR DESCRIPTION
**Pull request by jerryhxu**
Closes https://github.com/cloud-custodian/cloud-custodian/issues/9776

### CODE REVIEW

**Comment by kapilt** [comment]
context https://aws.amazon.com/about-aws/whats-new/2023/11/amazon-kinesis-data-streams-cross-account-access-aws-lambda/

File: c7n/resources/kinesis.py [hunk 1]
```diff
@@ -459,3 +460,33 @@ def process_resource_set(self, client, resource_set, tag_keys):
...
+
+    permissions = ('kinesis:GetResourcePolicy',)
+
+    def get_resource_policy(self, r):
+        client = local_session(self.manager.session_factory).client('kinesis')
+        result = self.manager.retry(
+                client.get_resource_policy,
+                ResourceARN=r['StreamARN'],
+                ignore_err_codes=('ResourceNotFoundException'))
+        policy = result.get(self.policy_attribute, None)
```
Comment (kapilt) [inline c7n/resources/kinesis.py:491]: re exception catch
```
if not result:
   return
```

Comment (jerryhxu) [reply c7n/resources/kinesis.py:496]: Good catch. I updated PR.

**Review by kapilt — APPROVED** [review]
lgtm

**Pull request merged**